GTK Native File Chooser support

fictitiousexistence opened the pull request:
Add support for using the File Chooser set by XDG Desktop Portal via    
"GTK_USE_PORTAL" environment variable.                                  
                                                                        
Users can change which File Chooser is used instead of using the default 
gtk chooser. For example, users can use kde's file chooser or even custom
choosers like: boydaihungst/xdg-desktop-portal-termfilechooser          
                                                                        
If GTK_USE_PORTAL is set to 0 or no alternate chooser is set the default 
gtk chooser will be used.                                               
                                                                        
Tested on sway,dwl,gnome,kde                                            
                                                                        
resolves #1107

fictitiousexistence commented:
Before applying this I noticed that after I downloaded anything I would get spammed with thousands of `[core/common/ipc]: Trying to send an ipc message, but the endpoint went away.` messages. This has not seemed to happen today.

balejk commented:
The error messages you are seeing are due to #1067, so they should not be related to your PR.

I haven't reviewed the actual code, just a nit: you seem to be making an unnecessary newline change in the middle hunk.

fictitiousexistence committed:
Add support for using the File Chooser set by XDG Desktop Portal

Users can change which File Chooser is used instead of using the default
gtk file chooser.
For example, users can use kde's file chooser or even custom
choosers like: boydaihungst/xdg-desktop-portal-termfilechooser

If GTK_USE_PORTAL is set to 0 or no alternate chooser is set, the default
gtk chooser will be used.
If GTK_USE_PORTAL is set to 1, the user specified file chooser will be used

Tested on sway,dwl,gnome,kde

resolves #1107

fictitiousexistence committed:
Merge branch 'develop' into GTKNativeFileChooser
